.github/workflows/pr_validation.yaml

```diff
@@ -26,3 +24,77 @@ jobs:
           echo "e.g. $PR_TITLE (BugFix)"
           exit 1
         fi
+
+  sru_gate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Checkbox monorepo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Install dependencies, Checkbox and providers
+        run: |
+          sudo apt install -y -qq python3 python3-venv jq libsystemd-dev
+          python3 -m venv venv
+          . venv/bin/activate
+          python3 -m pip install checkbox-ng/
+          python3 -m pip install checkbox-support/
+          python3 providers/resource/manage.py develop
+          python3 providers/base/manage.py develop
+          python3 providers/certification-client/manage.py develop
+          python3 providers/tpm2/manage.py develop
+          python3 providers/sru/manage.py develop
+      - name: Export main sru list
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR: ${{ github.event.pull_request.html_url }}
+        run: |
+          export DESCRIPTION=$(gh pr view $PR --json 'body' --jq '.body')
+          # we are on current branch/commit
+          . venv/bin/activate
+          checkbox-cli expand -f json com.canonical.certification::sru-server >> sru_server_testplan_this_branch.json
+          checkbox-cli expand -f json com.canonical.certification::sru >> sru_testplan_this_branch.json
+          git checkout origin/main
+          checkbox-cli expand -f json com.canonical.certification::sru-server >> sru_server_testplan_main.json
+          checkbox-cli expand -f json com.canonical.certification::sru >> sru_testplan_main.json
+          # https://stackoverflow.com/a/57795761
+          # compare two jsons without comparing the order arrays but just
+          # the content of objects
+
+          echo
+          echo
+          echo "Difference of com.canonical.certification::sru-server (diff this_branch main)"
+          diff \
+            <(jq -S 'def post_recurse(f): def r: (f | select(. != null) | r), .; r; def post_recurse: post_recurse(.[]?); (. | (post_recurse | arrays) |= sort)' "sru_server_testplan_this_branch.json") \
```

Comment:
Can you just add a comment somewhere to explain what these few lines of madness are actually doing?

Reply:
yup, of course this needed an explanation! fixed